cmd/rebasehelpers/util/git.go

```diff
@@ -0,0 +1,97 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type Commit struct {
+	Sha     string
+	Summary string
+	Files   []File
+}
+
+func (c Commit) DeclaresUpstreamChange() bool {
+	return strings.Contains(strings.ToLower(c.Summary), "upstream")
```

Comment:
We validate the case elsewhere: this is just to detect whether a commit summary "looks like an upstream commit," so we want to catch use of the term regardless of case.